AmplifyPlugins/DataStore/Tests/DataStoreHostApp/DataStoreStressTests/DataStoreStressTests.swift

```diff
@@ -0,0 +1,432 @@
+                    }
+
+                    if mutationEvent.mutationType == MutationEvent.MutationType.create.rawValue,
+                       mutationEvent.version == 1 {
+                        postsSyncedToCloudCount += 1
+                        await postsSyncedToCloud.fulfill()
+                    }
+                }
+            } catch {
+                XCTFail("Failed \(error)")
+            }
+        }
+
+        let capturedPosts = posts
+
+        DispatchQueue.concurrentPerform(iterations: concurrencyLimit) { index in
+            Task {
+                _ = try await Amplify.DataStore.save(capturedPosts[index])
+            }
+        }
+        
+        await waitForExpectations([postsSyncedToCloud], timeout: networkTimeout)
+    }
+    
+    /// Perform concurrent saves and observe the data successfuly synced from cloud
+    ///
+    /// - Given: DataStore is in ready state
+    /// - When:
+    ///    - Concurrently perform Datastore.save() from 50 tasks and Datastore.query(model:byId:) from 50 tasks
+    /// - Then:
+    ///    - Ensure the expected mutation event with version 1 (synced from cloud) is received
+    ///    - Query for the item is successful
+    ///
+    func testMultipleQueryByID() async throws {
+        await setUp(withModels: TestModelRegistration(), logLevel: .verbose)
+        try await startAmplifyAndWaitForSync()
+
+        var posts = [Post]()
+        for index in 0 ..< concurrencyLimit {
+            let post = Post(title: "title \(index)",
+                            content: "content",
+                            createdAt: .now())
+            posts.append(post)
+        }
+        let postsSyncedToCloud = asyncExpectation(description: "All posts saved and synced to cloud",
+                                                  expectedFulfillmentCount: concurrencyLimit)
+
+        let postsCopy = posts
+        Task {
+            var postsSyncedToCloudCount = 0
+            let mutationEvents = Amplify.DataStore.observe(Post.self)
+            do {
+                for try await mutationEvent in mutationEvents {
+                    guard postsCopy.contains(where: { $0.id == mutationEvent.modelId }) else {
+                        return
+                    }
+
+                    if mutationEvent.mutationType == MutationEvent.MutationType.create.rawValue,
+                       mutationEvent.version == 1 {
+                        postsSyncedToCloudCount += 1
+                        await postsSyncedToCloud.fulfill()
+                    }
+                }
+            } catch {
+                XCTFail("Failed \(error)")
+            }
+        }
+
+        let capturedPosts = posts
+
+        DispatchQueue.concurrentPerform(iterations: concurrencyLimit) { index in
+            Task {
+                _ = try await Amplify.DataStore.save(capturedPosts[index])
+            }
+        }
+        
+        await waitForExpectations([postsSyncedToCloud], timeout: networkTimeout)
```

Comment:
this code makes sense to be the body of `testMultipleSave`, but now it's duplicated in `testMultipleQueryByID` as the "set up" portion of the test. we can refactor this to something like `let savedAndSyncedPosts = try await saveAndSyncPosts(concurrenyLimit: concurrenyLimit)` for example